src/VSIX/NpgsqlConnectionEquivalencyComparer.cs

```diff
@@ -7,18 +8,39 @@
 
 namespace Npgsql.VSIX
 {
-    class NpgsqlConnectionEquivalencyComparer : IVsDataConnectionEquivalencyComparer
+    class NpgsqlConnectionEquivalencyComparer : DataConnectionEquivalencyComparer
     {
-        public bool AreEquivalent(string connectionString1, string connectionString2)
+        protected override bool AreEquivalent(IVsDataConnectionProperties connectionProperties1, IVsDataConnectionProperties connectionProperties2)
         {
-            NpgsqlConnectionStringBuilder parser1 = new NpgsqlConnectionStringBuilder(connectionString1);
-            NpgsqlConnectionStringBuilder parser2 = new NpgsqlConnectionStringBuilder(connectionString2);
-            return true
-                && parser1.Host.Equals(parser2.Host)
-                && parser1.Port.Equals(parser2.Port)
-                && parser1.Database.Equals(parser2.Database)
-                && parser1.Username.Equals(parser2.Username)
-                ;
+            return SummalizeConnectionProperties(connectionProperties1)
```

Comment:
Ok I have fixed it.

> But is there any specific reason to enumerate all the properties

Just I don't expect generous behavior of `IDictionary<string, object>` that may throw KeyNotFoundException on key absent...
But I have confirmed that nothing thrown in this case. It will be ok...